packages/py-api/py_api/controllers/hackathon/participants_controller.py

```diff
@@ -95,144 +85,63 @@ def delete_participant_from_team(cls, deleted_participant: Dict[str, Any]) -> Tu
-
-            updated_participant = ParticipantsFunctionality.update_participant(
-                id=str(inserted_participant.inserted_id), participant=participant,
-            )
-
-            if not updated_participant:
-                return JSONResponse(status_code=500, content={"message": "Something went wrong updating the participant with new team"})
-
-            return {"message": "Participant was successfully verified and appended to corresponding team"}
+            # The jwt token is provided as a query_param when a participant is registering via the custom form in
+            # the frontend
+            return await cls.handle_registration_via_invite_link(jwt_token, participant)
 
+        # The participant has provided a team name upon registration, so we assign them as admin to the newly
+        # created team
         if participant.team_name:
-            participant.is_verified = False
             participant.is_admin = True
+            return await cls.add_participant_to_new_team(participant)
 
-            inserted_participant = ParticipantsFunctionality.create_participant(
-                participant,
-            )
-
-            if not inserted_participant:
-                return JSONResponse(status_code=500, content={"message": "Failed inserting new participant"})
-
-            if TeamFunctionality.get_count_of_teams() > 15:
-                return JSONResponse(content={"message": "Hackathon is at max capacity"}, status_code=409)
-
-            team_object = TeamFunctionality.create_team_object_with_admin(
-                user_id=str(inserted_participant.inserted_id), team_name=participant.team_name,
-            )
-
-            if not team_object:
-                # or might be a bad transaction
-                return JSONResponse(content={"message": "Name is already taken"}, status_code=400)
-
-            new_team = TeamFunctionality.insert_team(team_object)
-
-            if not new_team:
-                # delete redundant participant document if team creation request has failed
-                ParticipantsFunctionality.delete_participant(
-                    str(inserted_participant.inserted_id),
-                )
-
-                return JSONResponse(status_code=500, content={"message": "Failed inserting new team. Participant entry was discarded."})
-
-            """
-                jwt_token = JWTFunctionality.create_jwt_token(
-                    participant.team_name,
-                )
-
-                * generate token and send verification link via email to the participant
-                # TODO: send_verification_email(to=participant.email, is_admin=True, verification_link=verification_link)
-
-            """
-
-            return {"message": "New admin participant was registered"}
-
+        # The participant hasn't provided a team name upon registration, so we follow the workflow for creating
+        # and adding random patricians to teams
         else:
```

Comment:
I am not sure if I should add participant.is_admin = True here as well

Reply:
I'm not sure which part of the workflow that is, but only the first member of a non-random team should be an admin.